crates/witgen_macro_helper/src/generator.rs

```diff
@@ -116,61 +116,80 @@ pub fn gen_wit_enum(enm: &ItemEnum) -> Result<String> {
-                } else {
-                    format!("{}({}),", variant_ident, fields)
-                };
+        .map(|variant| {
+            let ident = variant.ident.to_string().to_kebab_case();
+            let comment = get_doc_comment(&variant.attrs)?;
+            let variant_string = match &variant.fields {
+                syn::Fields::Named(_named) => {
+                    let fields = _named
+                        .named
+                        .iter()
+                        .map(|field| {
+                            field.ty.to_wit().map(|ty| {
+                                let field_doc = get_doc_comment(&field.attrs)
+                                    .unwrap_or(None)
+                                    .as_ref()
+                                    .map_or("".to_string(), |s| format!("    {s}"));
 
-                match comment {
-                    Some(comment) => Ok(format!("{}    {}", comment, variant_wit)),
-                    None => Ok(variant_wit),
+                                format!(
+                                    "{}    {}: {}",
+                                    field_doc,
+                                    field.ident.as_ref().unwrap(),
+                                    ty
+                                )
+                            })
+                        })
+                        .collect::<Result<Vec<String>>>()?
+                        .join(",\n");
+                    let inner_type_name = &format!("{enm_name}-{ident}");
+                    let comment = comment.as_deref().unwrap_or_default();
+                    named_types.push_str(&format!(
+                        "{comment}record {inner_type_name} {{\n{fields}\n}}\n"
+                    ));
+                    Ok(format!("{}({})", ident, inner_type_name))
                 }
-            }
-            syn::Fields::Unit => {
-                let comment = get_doc_comment(&variant.attrs)?;
-                let variant_wit = variant.ident.to_string().to_kebab_case() + ",";
+                syn::Fields::Unnamed(unamed) => {
+                    let fields = unamed
+                        .unnamed
+                        .iter()
+                        .map(|field| field.ty.to_wit())
+                        .collect::<Result<Vec<String>>>()?
+                        .join(", ");
+                    is_known_keyword(&ident)?;
 
-                match comment {
-                    Some(comment) => Ok(format!("{}    {}", comment, variant_wit)),
-                    None => Ok(variant_wit),
+                    Ok(if unamed.unnamed.len() > 1 {
+                        format!("{}(tuple<{}>),", ident, fields)
+                    } else {
+                        format!("{}({}),", ident, fields)
+                    })
                 }
-            }
+                syn::Fields::Unit => Ok(ident + ","),
+            };
+            let comment = comment.map_or("".to_string(), |s| format!("    {s}"));
```

Comment:
If we're using this syntax `"{s}"` I think we will need to set the https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field at 1.58. Or we decide to not already use this syntax to be compliant with lower versions

Reply:
Ah good point. I'll revert to using the old way.  I think it'd be better to do a full upgrade in a later PR.